src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionListener.cs

```diff
@@ -26,6 +26,11 @@ internal class QuicConnectionListener : IMultiplexedConnectionListener, IAsyncDi
 
         public QuicConnectionListener(QuicTransportOptions options, IQuicTrace log, EndPoint endpoint, SslServerAuthenticationOptions sslServerAuthenticationOptions)
         {
+            if (!QuicImplementationProviders.Default.IsSupported)
+            {
+                throw new NotSupportedException("Quic is not supported or enabled on this platform.");
```

Comment:
nit: Lets refer to it to QUIC in exceptions and docs.
```suggestion
                throw new NotSupportedException("QUIC is not supported or enabled on this platform.");
```